robotoff/insights/importer.py

```diff
@@ -2106,24 +2106,43 @@ def import_product_predictions(
 
     deleted = 0
     if delete_previous_versions:
-        for (
-            server_type,
-            prediction_type,
-            source_image,
-            predictor_version,
-        ), _ in itertools.groupby(
-            sorted(product_predictions, key=_import_product_predictions_sort_fn),
-            _import_product_predictions_sort_fn,
-        ):
+    sorted_predictions = sorted(
+        product_predictions,
+        key=_import_product_predictions_sort_fn,
+    )
+
+    for (
+        server_type,
+        prediction_type,
+        source_image,
+        predictor_version,
+    ), _ in itertools.groupby(
+        sorted_predictions,
+        key=_import_product_predictions_sort_fn,
+    ):
+        if prediction_type.name == "category":
+            # Delete all previous category predictions
+            deleted += (
+                PredictionModel.delete()
+                .where(
+                    SQL(
+                        "prediction.barcode = %s AND "
+                        "prediction.server_type = %s AND "
+                        "prediction.type = %s",
+                        (
+                            barcode,
+                            server_type.name,
+                            prediction_type.name,
+                        ),
+                    )
```

Comment:
You don't need to use SQL here, you can use standard peewee syntax.
```suggestion
                    PredictionModel.barcode == barcode,
                    PredictionModel.server_type == server_type.name,
                    PredictionModel.type == prediction_type.name
```